### PR TITLE
BXMSPROD-1265: Conscientious language initiative [master]

### PR DESCRIPTION
**Pull request by mbiarnes**
**Thank you for submitting this pull request**

**JIRA**: [BXMSPROD-1265](https://issues.redhat.com/browse/BXMSPROD-1265)

**referenced Pull Requests**:
https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1704
https://github.com/kiegroup/kie-jenkins-scripts/pull/1040
https://github.com/kiegroup/jenkins-pipeline-shared-libraries/pull/160

<pre>
How to retest a PR or trigger a specific build:

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
</pre>

### CODE REVIEW

**Comment by mbiarnes** [comment]
srcclr_scan_job.groovy failed cause of:
https://github.com/kiegroup/kie-jenkins-scripts/pull/1040/files#diff-9d55ce9075a526a3684617ae54e1f15ee81673ba5c89794b916086a8ffacc4d9R55
`main` is not yet existing

**Review by mareknovotny — APPROVED** [review]

File: job-dsls/jobs/kie/kogito/kogito_docs_upload.groovy [hunk 1]
```diff
@@ -167,7 +167,7 @@ pipeline {
 pipelineJob("${folderPath}/uploadKogitoDocs") {
 
     description('''this job <b>uploadKogitoDocs</b>: <br><br>
-1. clones kie-docs - <b>“master-kogito”</b>  branch <br>
+1. clones kie-docs - <b>“main-kogito”</b>  branch <br>
```
Comment (Ginxo) [inline job-dsls/jobs/kie/kogito/kogito_docs_upload.groovy:170]:
```suggestion
1. clones kie-docs - <b>“master-kogito”</b>  branch <br>
```

File: job-dsls/jobs/kie/kogito/kogito_docs_upload.groovy [hunk 2]
```diff
@@ -179,7 +179,7 @@ pipelineJob("${folderPath}/uploadKogitoDocs") {
 ''')
 
     parameters {
-        stringParam("currentKieSnapshot","${currentKieSnapshot}","please enter the <b>current kie snapshot version</b> in poms of <b>master-kogito</b> branch<br>look at <br> https://github.com/kiegroup/kie-docs/blob/master-kogito/pom.xml OR <br>https://github.com/kiegroup/kie-docs/blob/master-kogito/doc-content/pom.xml OR <br>https://github.com/kiegroup/kie-docs/blob/master-kogito/doc-content/kogito-docs/pom.xml#L9")
+        stringParam("currentKieSnapshot","${currentKieSnapshot}","please enter the <b>current kie snapshot version</b> in poms of <b>main-kogito</b> branch<br>look at <br> https://github.com/kiegroup/kie-docs/blob/main-kogito/pom.xml OR <br>https://github.com/kiegroup/kie-docs/blob/main-kogito/doc-content/pom.xml OR <br>https://github.com/kiegroup/kie-docs/blob/main-kogito/doc-content/kogito-docs/pom.xml#L9")
```
Comment (Ginxo) [inline job-dsls/jobs/kie/kogito/kogito_docs_upload.groovy:182]:
```suggestion
        stringParam("currentKieSnapshot","${currentKieSnapshot}","please enter the <b>current kie snapshot version</b> in poms of <b>master-kogito</b> branch<br>look at <br> https://github.com/kiegroup/kie-docs/blob/master-kogito/pom.xml OR <br>https://github.com/kiegroup/kie-docs/blob/master-kogito/doc-content/pom.xml OR <br>https://github.com/kiegroup/kie-docs/blob/master-kogito/doc-content/kogito-docs/pom.xml#L9")
```

File: job-dsls/jobs/tools/tools_slave_image_build_PR_test.groovy [hunk 3]
```diff
@@ -54,7 +54,7 @@ job(jobName) {
 
     parameters {
         stringParam ("OWNER","kiegroup","")
-        stringParam ("BRANCH","master","")
+        stringParam ("BRANCH","main","")
```
Comment (Ginxo) [inline job-dsls/jobs/tools/tools_slave_image_build_PR_test.groovy:57]:
```suggestion
        stringParam ('BRANCH','main','')
```

File: job-dsls/jobs/tools/tools_slave_image_build_PR_test.groovy [hunk 4]
```diff
@@ -85,7 +85,7 @@ job(jobName) {
                 url('ssh://jb-ip-tooling-jenkins@code.engineering.redhat.com/bxms-jenkins')
                 credentials('code.engineering.redhat.com')
             }
-            branch ("master")
+            branch ("main")
```
Comment (Ginxo) [inline job-dsls/jobs/tools/tools_slave_image_build_PR_test.groovy:88]:
```suggestion
            branch ('main')
```

File: job-dsls/src/main/groovy/org/kie/jenkins/jobdsl/templates/PrVerificationJob.groovy [hunk 5]
```diff
@@ -143,9 +143,9 @@ class PrVerificationJob extends BasicJob {
                     useGitHubHooks(true)
 
                     // Adding branches to this whitelist allows you to selectively test pull requests destined for these branches only.
-                    // Supports regular expressions (e.g. 'master', 'feature-.*').
+                    // Supports regular expressions (e.g. 'main', 'feature-.*').
                     if (projectName == "kie-docs") {
-                        whiteListTargetBranches([branchName, "master-kogito"])
+                        whiteListTargetBranches([branchName, "main-kogito"])
```
Comment (Ginxo) [inline job-dsls/src/main/groovy/org/kie/jenkins/jobdsl/templates/PrVerificationJob.groovy:148]:
```suggestion
                        whiteListTargetBranches([branchName, "master-kogito"])
```

**Review by Ginxo — CHANGES_REQUESTED** [review]
please check my comments

**Review by Ginxo — APPROVED** [review]
LGTM, just few minor suggestions

File: job-dsls/jobs/kie/kie-tools/send_UMB_trigger_after_version_upgrade.groovy [hunk 6]
```diff
@@ -22,7 +22,7 @@ pipeline {
         stage ('UMB message'){
             steps {
                 sendCIMessage failOnError: false, messageContent: "\${VERSION_ORG_KIE}", messageProperties: '''CI_TYPE=custom
-label=rhba-ci''', messageType: 'Custom', overrides: [topic: 'VirtualTopic.qe.ci.ba.rhba.master.update-versions'], providerName: 'Red Hat UMB'  
+label=rhba-ci''', messageType: 'Custom', overrides: [topic: 'VirtualTopic.qe.ci.ba.rhba.main.update-versions'], providerName: 'Red Hat UMB'
```
[the remark this answers is not in the document]
Comment (winklerm) [reply job-dsls/jobs/kie/kie-tools/send_UMB_trigger_after_version_upgrade.groovy:25]: @cimbalek FYI ^

File: job-dsls/jobs/prod/prod_kogito_properties_generator.groovy [hunk 7]
```diff
@@ -61,8 +61,8 @@ pipelineJob("${folderPath}/kogito-properties-generator") {
 
     parameters {
         booleanParam("IS_PROD", true, "it defines if the properties file is for prod or not")
-        stringParam("BRANCH_NAME", "master", "the branch the nightly was triggered for")
-        stringParam("REPO_URL", "http://bxms-qe.rhev-ci-vms.eng.rdu2.redhat.com:8081/nexus/content/repositories/kogito-master-nightly", "Prod possibility is http://rcm-guest.app.eng.bos.redhat.com/rcm-guest/staging")
+        stringParam("BRANCH_NAME", "main", "the branch the nightly was triggered for")
+        stringParam("REPO_URL", "http://bxms-qe.rhev-ci-vms.eng.rdu2.redhat.com:8081/nexus/content/repositories/kogito-main-nightly", "Prod possibility is http://rcm-guest.app.eng.bos.redhat.com/rcm-guest/staging")
```
Comment (winklerm) [inline job-dsls/jobs/prod/prod_kogito_properties_generator.groovy:65]: Has this been renamed/created in Nexus?

File: job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy [hunk 8]
```diff
@@ -75,7 +75,7 @@ pipeline{
         stage('trigger kogito-tooling nightly job main') {
             steps {
                 build job: 'kogito-tooling.nightly/main', propagate: false, wait: true, parameters: [
-                        [\$class: 'StringParameterValue', name: 'DEPLOYMENT_REPO_URL', value: 'https://bxms-qe.rhev-ci-vms.eng.rdu2.redhat.com:8443/nexus/service/local/repositories/scratch-release-rhba-master/content-compressed'],
+                        [\$class: 'StringParameterValue', name: 'DEPLOYMENT_REPO_URL', value: 'https://bxms-qe.rhev-ci-vms.eng.rdu2.redhat.com:8443/nexus/service/local/repositories/scratch-release-rhba-main/content-compressed'],
```
Comment (winklerm) [inline job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy:78]: Has this been renamed/created in Nexus along with the scheduled task for clean up?

We might need to adjust the scheduled task ID in QE jobs.

**Review by winklerm — APPROVED** [review]
Left some comments regarding the pre-requisites.

File: job-dsls/jobs/prod/prod_kogito_properties_generator.groovy (hunk 7 above)
Comment (mbiarnes) [reply job-dsls/jobs/prod/prod_kogito_properties_generator.groovy:65]: yes -it has been created: 
https://bxms-qe.rhev-ci-vms.eng.rdu2.redhat.com:8443/nexus/content/groups/kogito-main-nightly/

File: job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy (hunk 8 above)
Comment (mbiarnes) [reply job-dsls/jobs/prod/prod_cron_meta_nightly_pipeline.groovy:78]: https://bxms-qe.rhev-ci-vms.eng.rdu2.redhat.com:8443/nexus/content/repositories/scratch-release-rhba-main/ ??

File: job-dsls/jobs/kie/kogito/kogito_docs_upload.groovy [hunk 9]
```diff
@@ -38,11 +38,11 @@ pipeline {
         }
         stage ('checkout kie-docs') {
             steps {
-                checkout([$class: 'GitSCM', branches: [[name: 'master-kogito']], browser: [$class: 'GithubWeb', repoUrl: 'git@github.com:kiegroup/kie-docs.git'], doGenerateSubmoduleConfigurations: false, extensions: [[$class: 'RelativeTargetDirectory', relativeTargetDir: 'kie-docs']], submoduleCfg: [], userRemoteConfigs: [[credentialsId: 'kie-ci-user-key', url: 'git@github.com:kiegroup/kie-docs.git']]])
+                checkout([$class: 'GitSCM', branches: [[name: 'main-kogito']], browser: [$class: 'GithubWeb', repoUrl: 'git@github.com:kiegroup/kie-docs.git'], doGenerateSubmoduleConfigurations: false, extensions: [[$class: 'RelativeTargetDirectory', relativeTargetDir: 'kie-docs']], submoduleCfg: [], userRemoteConfigs: [[credentialsId: 'kie-ci-user-key', url: 'git@github.com:kiegroup/kie-docs.git']]])
```
Comment (Ginxo) [inline job-dsls/jobs/kie/kogito/kogito_docs_upload.groovy:41]: this requires https://github.com/kiegroup/kie-docs/blob/d7ed223857d1e18588c76802eb765137140123ed/.github/workflows/push.yml#L7 to be updated

**Review by Ginxo — CHANGES_REQUESTED** [review]
please check my comments

File: job-dsls/jobs/kie/kogito/kogito_docs_upload.groovy (hunk 9 above)
Comment (mbiarnes) [reply job-dsls/jobs/kie/kogito/kogito_docs_upload.groovy:41]: @Ginxo good spot:
https://github.com/kiegroup/kie-docs/pull/3806

**Comment by Ginxo** [comment]
@mbiarnes sorry but I'm little by little discovering things:
* the `master` from https://github.com/kiegroup/optaplanner/blob/2ba8f2ac3b0b0e511219cc34f7a67c26f5119d5c/.github/workflows/pull_request.yml#L37
* https://github.com/kiegroup/optaweb-employee-rostering/blob/83d1ea6a805535a836fab0b41e6d8660ac108c4a/.github/workflows/pull_request.yml#L41
* https://github.com/kiegroup/optaweb-employee-rostering/blob/198e5996f07689f1a727b0e97806a38e29363816/.github/workflows/pull_request.yml#L40
* https://github.com/kiegroup/optaweb-vehicle-routing/blob/c28687b86e379b53e26bcdb7a7bffcd01a508903/.github/workflows/pull_request.yml#L46
* https://github.com/kiegroup/optaweb-vehicle-routing/blob/4fe642ea8ce53840f99a8ac03138ff068670f586/.github/workflows/pull_request.yml#L40

**Pull request merged**